packages/coreutils/src/poll.ts

```diff
@@ -11,12 +11,12 @@ import { ISignal, Signal } from '@phosphor/signaling';
  * A readonly poll that calls an asynchronous function with each tick.
  *
  * @typeparam T - The resolved type of the factory's promises.
- * Defaults to `any`.
  *
  * @typeparam U - The rejected type of the factory's promises.
- * Defaults to `any`.
+ *
+ * @typeparam V - The type to extend the phases supported by a poll.
  */
-export interface IPoll<T = any, U = any> {
+export interface IPoll<T, U, V extends string> {
```

Comment:
This way of extending the poll phases feels off to me. What if the user wants to extend with more than one phase? Is there a reason this generic is not `V extends IPoll.State<T,U>`, so that the user can extend with any new phases?

Reply:
You can add multiple phases by doing `IPoll<T, U, 'foo | 'bar' | 'baz'>`

Reply:
Ah, right, makes sense.